src/compiler/binder.ts

```diff
@@ -3340,6 +3341,9 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
                 }
             });
         }
+        if (isPrototypeProperty && namespaceSymbol && namespaceSymbol.valueDeclaration) {
+            if (isAssignmentDeclaration(namespaceSymbol.valueDeclaration)) containerIsClass = false;

```

Comment:
Nit, but:

```suggestion
            if (isAssignmentDeclaration(namespaceSymbol.valueDeclaration)) {
                containerIsClass = false;
            }
```

In general my impression is that we avoid one-liner things like this unless they are just `return` or `break` or something

Reply:
Also consider just smushing the to `if`s together.